Objects/genobject.c

```diff
@@ -704,7 +705,8 @@ static PyObject *
 gen_get_name(PyObject *self, void *Py_UNUSED(ignored))
 {
     PyGenObject *op = _PyGen_CAST(self);
-    return Py_NewRef(op->gi_name);
+    PyObject *name = FT_ATOMIC_LOAD_PTR_RELAXED(op->gi_name);
+    return Py_NewRef(name);
```

Comment:
Let's use "acquire" here. You could make an argument that "relaxed" is good enough (e.g., 2.6.7 in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2055r0.pdf), but:

1) "acquire" is more conformant with the c11 memory model
2) the perf difference is unlikely to matter (same instructions on x86-64 and negligible overhead on apple's arm processors)
3) it avoids potential problems reported by tsan

```suggestion
    PyObject *name = FT_ATOMIC_LOAD_PTR_ACQUIRE(op->gi_name);
    return Py_NewRef(name);
```